shpc/main/registry/provider.py

```diff
@@ -33,7 +33,7 @@ class Provider:
     """
 
     def __init__(self, source, *args, **kwargs):
-        if not (source.startswith("https://") or os.path.exists(source)):
+        if not (source.startswith("https://") or source.startswith("ssh://") or os.path.exists(source)):
```

Comment:
I agree with you here this could be more elegant - what if we have more of a regular expression that is assembled to match any of those potential protocols and then properly does the check/replacement?